interface/github.py

```diff
@@ -1,80 +1,90 @@
 """Utility classes for interacting with Github API via PyGithub."""
-from github import GithubObject, GithubException
+from github import Github, GithubObject, GithubException
+from interface.exceptions.github import GithubAPIException
+from interface.github_app import GithubAppInterface, \
+    DefaultGithubAppAuthFactory
+
+
+def handle_github_error(func):
+    """Github error handler that updates Github App API token if necessary."""
+    def wrapper(self, *arg, **kwargs):
+        try:
+            return func(self, *arg, **kwargs)
+        except GithubException as e:
+            if e.status == 401:
+                self.github = self.github_factory.create()
+                try:
+                    return func(self, *arg, **kwargs)
+                except GithubException as e:
+                    raise GithubAPIException(e.data)
+            raise GithubAPIException(e.data)
```

Comment:
this works but would be clearer if you added an `else` statement

Reply:
Fixed.